supervisord.conf

```diff
@@ -27,6 +27,17 @@ stdout_logfile = /var/log/mysql.log
 stderr_logfile = /var/log/mysql.log
 autorestart = true
 
+[program:postgresql]
+priority = 1
+command =
+    /usr/pgsql-10/bin/postgres
+        -D /srv/postgres
+        -r /srv/logs/postgres.log
+stdout_logfile = /srv/logs/postgres.log
+stderr_logfile = /srv/logs/postgres.log
+user=postgres
+autorestart=true
```

Comment:
> To terminate the postgres server normally, the signals SIGTERM, SIGINT, or SIGQUIT can be used. The first will wait for all clients to terminate before quitting, the second will forcefully disconnect all clients, and the third will quit immediately without proper shutdown, resulting in a recovery run during restart.

We need `SIGINT`, but supervisord uses `SIGTERM` by default. We need to change that option.

Reply:
The stop signal has been changed.
Here are logs of my tests
---
2018-12-21 09:54:24,806 DEBG killing postgresql (pid 20) with signal SIGINT
2018-12-21 09:54:24,834 INFO stopped: postgresql (exit status 0)
---